yt/fields/astro_fields.py

```diff
@@ -145,7 +145,7 @@ def _four_velocity(field, data):
 
         return _four_velocity
 
-    for u in "xyz":
+    for u in registry.ds.coordinates.axis_order:
```

Comment:
great catch

Reply:
+1

Reply:
although... I'd like to be sure what to expect. Right now the `T_mu_nu` fields you're adding are only compatible with cartesian geometries, right ? If so, this change still has value because it's one less obstacle to overcome when/if we want to extend support to other geometries in the future, but it doesn't do it just by itself, right ?

Reply:
This change would add the `four_velocity` fields to non-Cartesian datasets now, but not the tensor, no

Reply:
then I guess I'm alright with it